pyo3-macros-backend/src/pyfunction/signature.rs

```diff
@@ -556,4 +564,67 @@ impl<'a> FunctionSignature<'a> {
             attribute: None,
         }
     }
+
+    pub fn text_signature(&self, self_argument: Option<&str>) -> String {
+        let mut output = String::new();
+        output.push('(');
+
+        if let Some(arg) = self_argument {
+            output.push('$');
+            output.push_str(arg);
+        }
+
+        let mut first = self_argument.is_none();
```

Comment:
Hiding the `first` binding would feel more robust to me, e.g.

```rust
let mut maybe_push_comma = {
    let mut first = self_argument.is_none();

    |output: &mut String| {
        if !first {
            output.push_str(", ");
        } else {
            first = false;
        }
    }
};
```

Reply:
Good idea 👍